spec/setup/helpers/abs_helper_spec.rb

```diff
@@ -0,0 +1,779 @@
+
+          expect(subject.abs_get_token_from_fog_file).to eq(nil)
+        end
+
+      end
+
+    end
+
+    context 'when the .fog file does not exist' do
+
+      it 'reports the error and returns nil' do
+        expect(File).to receive(:exist?).and_return(false)
+        expect(subject).to receive(:puts).with(/fog file not found/).at_least(:once)
+
+        expect(subject.abs_get_token_from_fog_file).to eq(nil)
+      end
+
+    end
+
+  end
+
+  describe '#abs_get_token' do
+
+    context 'when a token is set as an environment variable' do
+
+      it 'returns the token' do
+        ENV['ABS_TOKEN'] = TEST_ABS_TOKEN
+        expect(subject.abs_get_token).to eq(TEST_ABS_TOKEN)
+      end
+
+    end
+
+    context 'when a token is not set as an environment variable' do
+
+      context 'when a token is set in the .fog file' do
+
+        it 'returns the token' do
+          ENV['ABS_TOKEN'] = nil
+          expect(subject).to receive(:abs_get_token_from_fog_file).and_return(TEST_ABS_TOKEN)
+          expect(subject.abs_get_token).to eq(TEST_ABS_TOKEN)
+        end
+
+      end
+
+      context 'when a token is not set in the .fog file' do
+
+        it 'reports the error and returns nil' do
+
+          ENV['ABS_TOKEN'] = nil
+          expect(subject).to receive(:abs_get_token_from_fog_file).and_return(nil)
+          expect(subject).to receive(:puts).with(/An ABS token must be set/)
+          expect(subject.abs_get_token).to eq(nil)
+        end
+
+      end
+
+    end
+
+  end
+
+  describe '#abs_get_request_post' do
+
+    context 'when an abs token is present' do
+
+      it 'returns the expected result' do
+        expect(subject).to receive(:abs_get_token).and_return(TEST_ABS_TOKEN)
+
+        stub_const('Net::HTTP::Post', test_net_http_post)
+        expect(test_net_http_post).to receive(:new).with(TEST_AWSDIRECT_URI, TEST_EXPECTED_REQUEST_HEADERS).and_return(test_http_request)
+
+        expect(subject.abs_get_request_post(TEST_AWSDIRECT_URI)).to eq(test_http_request)
+      end
+
+    end
+
+    context 'when an abs token is not present' do
+
+      it 'reports the error and returns nil' do
+        test_expected_message = 'Unable to prepare a valid ABS request without a valid token'
+        test_expected_result = nil
+
+        expect(subject).to receive(:abs_get_token).and_return(nil)
+        expect(subject).to receive(:puts).with(test_expected_message)
+
+        expect(subject.abs_get_request_post(TEST_AWSDIRECT_URI)).to eq(test_expected_result)
+      end
+
+    end
+
+  end
+
+  describe '#abs_request_awsdirect' do
+
+    context 'when a request is successfully prepared' do
+
+      it 'performs the request, reports and returns the response' do
+        expect(subject).to receive(:abs_get_request_post).and_return(test_http_request)
+        allow(test_http_request).to receive(:body=)
+
+        stub_const('Net::HTTP', test_net_http)
+        allow(test_net_http).to receive(:new).and_return(test_net_http_instance)
```

Comment:
on line 356 you did basically the same thing, but used expect, whereas here you used allow. Was that intentional?

Reply:
Good catch; I've updated this to use expect.